packages/component/src/hooks/useEmojiFromStyles.js

```diff
@@ -0,0 +1,51 @@
+/*eslint no-useless-escape: "off"*/
+/*eslint require-unicode-regexp: "off" */
+import useStyleOptions from './useStyleOptions';
+
+function escapeRegexp(emoticon) {
+  return emoticon.replace(/[\\^$*+?.()|[\]{}]/g, '\\$&');
+}
+
+export default function useEmojiFromStyles() {
+  const [{ emojiAutocorrect: autocorrect, emojiList: customEmojiList }] = useStyleOptions();
+
+  const emojiAutocorrect = autocorrect || false;
```

Comment:
1. Please do all the patching in `Composer`. By patching, I mean, in this line, you are patching `autocorrect` to become truthy/`false`
   - This is because other components may call `useStyleOptions` and then, get an unpatched value
   - Patching everything as close to the source as possible will reduce complexity. Source means, the props passed by the dev.
1. Instead of truthy/`false`, do you think it is more appropriate for `true`/`false`?